gateway/response.go

```diff
@@ -91,6 +91,15 @@ func (fw *ResponseForwarder) ForwardMessage(ctx context.Context, mux *runtime.Se
 	}
 
 	retainFields(ctx, req, dynmap)
+	errs, suc := errorsAndSuccessFromContext(ctx)
+	if _, ok := dynmap["error"]; len(errs) > 0 && !ok {
+		dynmap["error"] = errs
+	}
+	// this is the edge case, if user sends response that has field 'success'
+	// let him see his response object instead of our status
+	if _, ok := dynmap["success"]; !ok && suc != nil {
+		dynmap["success"] = suc
+	}
```

Comment:
There's a tradeoff there, either every `forward_{Service}_{RPC}_#` has to be manually overridden to `gateway.ForwardResponseMessage`, or every RPC response message type has to explicitly include the `success` and/or `error` fields. Right now the former seems more palatable to me because that requires no change from the current, although the message approach probably would've been better from the get-go. I could see if there's a way to support both.

Reply:
Makes sense. It would be ugly to force all the clients to add `success` and `error` to the response.